src/ladds/distributedMemParallelization/AltitudeBasedDecomposition.cpp

```diff
@@ -0,0 +1,272 @@
+/**
+ * @file AltitudeBasedDecomposition.cpp
+ * @author P. Gomez
+ * @date 14.07.22
+ */
+
+#include "AltitudeBasedDecomposition.h"
+
+#include <autopas/utils/ArrayMath.h>
+#include <autopas/utils/WrapOpenMP.h>
+#include <spdlog/spdlog.h>
+
+#include "ladds/TypeDefinitions.h"
+#include "ladds/distributedMemParallelization/ParticleCommunicator.h"
+#include "satellitePropagator/physics/Constants.h"
+
+LADDS::AltitudeBasedDecomposition::AltitudeBasedDecomposition(LADDS::ConfigReader &config) {
+  // initialize a one-dimensional MPI world in which split the altitudes
+  int numRanks{};
+  autopas::AutoPas_MPI_Comm_size(AUTOPAS_MPI_COMM_WORLD, &numRanks);
+  // let MPI decide a dimension splitting with no periodicity
+  std::array<int, 1> periods{false};
+  std::array<int, 1> dims{};
+  autopas::AutoPas_MPI_Dims_create(numRanks, dims.size(), dims.data());
+  // create a communicator
+  autopas::AutoPas_MPI_Cart_create(
+      AUTOPAS_MPI_COMM_WORLD, dims.size(), dims.data(), periods.data(), true, &communicator);
+  // get coordinates of this rank within the decomposition grid
+  int rank{};
+  autopas::AutoPas_MPI_Comm_rank(communicator, &rank);
+  std::array<int, 1> coords{};
+  autopas::AutoPas_MPI_Cart_coords(communicator, rank, dims.size(), coords.data());
+
+  // initialize simulation specific data
+  const auto maxAltitude = config.get<double>("sim/maxAltitude");
+  const auto minAltitude = config.get<double>("sim/minAltitude");
+
+  globalBoxMin = {-maxAltitude, -maxAltitude, -maxAltitude};
+  globalBoxMax = {maxAltitude, maxAltitude, maxAltitude};
+
+  // calculate local box extent
+  // we have to account for the fact that there won't be anything between 0, Physics::R_EARTH  + minAltitude, so
+  // shifting the interval a bit
+  // TODO: find a more even distribution of the particles
```

Comment:
want to turn this into an issue or is this already good enough?

Reply:
Already implemented here https://github.com/esa/LADDS/blob/b3eef8f1252486c45848877f94eda1e1efb75a74/src/ladds/distributedMemParallelization/AltitudeBasedDecomposition.cpp#L132 , removed the todo